File: code/basecode/python/scene.py
from utils import jsonfile2obj

def loadScene(sceneFile):
    return jsonfile2obj(sceneFile, object_hook=sceneHook)


def sceneHook(obj):
    if '__type__' in obj:
        if obj['__type__'] == "scene":
            #ToDo
        if obj['__type__'] == "camera":
			fov = obj["fov"]
			position = obj["position"]
        if obj['__type__'] == "sphere":
            #ToDo
        if obj['__type__'] == "point_light":
            #ToDo        
    return obj

File: code/basecode/c#/Scene.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using Newtonsoft.Json.Linq;

namespace raytracer
{
  class Scene
  {
    public static void LoadScene(string fileName)
    {
      var jsonString = File.ReadAllText(fileName);
      ObjectHook(JToken.Parse(jsonString));      
    }

    private static object ObjectHook(JToken token)
    {
      switch (token.Type)
      {
        case JTokenType.Object:

          var children = token.Children<JProperty>();
          var dic = children
                      .ToDictionary(prop => prop.Name,
                                    prop => ObjectHook(prop.Value));
          if (dic.ContainsKey("__type__"))
          {
            if (dic["__type__"].ToString() == "scene")
            {
              //ToDo
            }
            else if (dic["__type__"].ToString() == "camera")
            {
              var fov = Convert.ToSingle(dic["fov"]);
              var position = ((List<object>)dic["position"]).Select(Convert.ToSingle).ToList();
			  //ToComplete
            }
            else if (dic["__type__"].ToString() == "sphere")
            {
              //ToDo
            }
            else if (dic["__type__"].ToString() == "point_light")
            {
              //ToDo
            }
          }
          return dic;

        case JTokenType.Array:
          return token.Select(ObjectHook).ToList();

        default:
          return ((JValue)token).Value;
      }
    }
  }
}
